add in-place macros @transform! and @select! (#216)
JuliaData/DataFramesMeta.jl: docs/src/index.md

Metaprogramming tools for DataFrames.jl objects to provide more convenient syntax.

> DataFrames.jl has the functions `select(!)`, `transform(!)`, and `combine` 

Reviewer: ```suggestion
DataFrames.jl has the functions `select`, `transform`, and `combine`, as well as the in-place `select!` and `transform!`
```

for manipulating data frames. DataFramesMeta provides the macros 

> `@select(!)`, `@transform(!)`, and `@combine` to mirror these functions with 

Reviewer: ```suggestion
`@select`, `@transform`, `@combine`, `@select!`, and `@transform!` to mirror these functions with 
```

@select(df, x2 = 2 * :x, :y)
@select(gd, x2 = 2 .* :y .* first(:y))
```

> ## `@select!`

Author: good idea. Combined in the latest commit.

@select(df, :x, :y)
@select(df, x2 = 2 * :x, :y)
@select(gd, x2 = 2 .* :y .* first(:y))
@select!(df, :x, :y)
@select!(df, x = 2 * :x, :y)
@select!(gd, y = 2 .* :y .* first(:y))
```

## `@transform and @transform!`

Add additional columns based on keyword arguments. Operates on both a 
`DataFrame` and a `GroupedDataFrame`. 

> `@transform` copies the original dataframe and return it together with newly created
> columns, while `@transform!` adds additional columns to the original dataframe and return it.

Reviewer 2: ```suggestion
`@transform` returns a new data frame with newly allocated columns, while `@transform!`
mutates the original data frame and returns it.
```
